src/Adapter/Attachment/QueryHandler/GetAttachmentHandler.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Attachment\QueryHandler;
+
+use PrestaShop\PrestaShop\Adapter\Attachment\AbstractAttachmentHandler;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Exception\AttachmentNotFoundException;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\Query\GetAttachment;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\QueryHandler\GetAttachmentHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Attachment\QueryResult\Attachment;
+
+/**
+ * Provides path and original file name of attachment
+ */
+final class GetAttachmentHandler extends AbstractAttachmentHandler implements GetAttachmentHandlerInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @throws AttachmentNotFoundException
+     */
+    public function handle(GetAttachment $query): Attachment
+    {
+        $attachment = $this->getAttachment($query->getAttachmentId());
+        $path = _PS_DOWNLOAD_DIR_ . $attachment->file;
+
+        if (!file_exists($path)) {
+            throw new AttachmentNotFoundException('Attachment file was not found');
```

Comment:
```suggestion
            throw new AttachmentNotFoundException(sprintf('Attachment file was not found at %s', $path));
```

Exceptions are meant to be read by developers and consequently it's best to provide as much as possible info to them. If file does not exists, developer will need the `$path` to understand where it's supposed to be. For example if `_PS_DOWNLOAD_DIR_` is missing a slash, he'll see it immediately this way 😉